[py2py3] Check unittests from Slice 12 in py3

#### Intro

We need to make sure that the work done in the migration slices
is py3 compatible. In particular, we should make sure that the
unittests that are involved in each slice do not fail.

The schedule for this validation / bug-fixing campaign is kept at #10422

#### Details

This issue is about checking that the unittests from Slice 12,
issue #10136 and PR #10307, do not fail.
A non-exhaustive list of the ut that need to pass is

```
test/python/WMCore_t/WorkQueue_t/WorkQueue_t.py
```

manually added, files have been renamed in #10357. there is no need to wait until slice 25 to fix these, though
```
test/python/WMCore_t/MicroService_t/MSOutput_t/MSOutputTemplate_t.py
test/python/WMCore_t/MicroService_t/MSRuleCleaner_t/MSRuleCleanerWflow_t.py
test/python/WMCore_t/MicroService_t/MSTransferor_t/Workflow_t.py
```

#### analysis of the failures

Only failures in py2
1. `test/python/WMCore_t/WorkQueue_t/WorkQueue_t.py:WorkQueueTest.testWMBSInjectionStatus`
2.  `test/python/WMCore_t/WorkQueue_t/WorkQueue_t.py:WorkQueueTest.testGlobalBlockSplitting`


broken py3: 

- [ ] `testWhiteList` - F - `line 566, in testWhiteList
    self.assertEqual(len(work), 17 + 19)
AssertionError: 28 != 36`
- [ ] `testProcessingWithContinuousSplitting` - F - `self.assertEqual(len(self.localQueue), 35)
AssertionError: 30 != 35`
- [ ] `test/python/WMCore_t/WorkQueue_t/WorkQueue_t.py:WorkQueueTest.testProcessing` : FAIL `line 500, in testProcessing
    self.assertEqual(processedBlocks, 17)
AssertionError: 12 != 17`

- [ ] `testBlackList` - `line 546, in testBlackList
    self.assertEqual(len(work), 17 + 19)
AssertionError: 33 != 36`

@amaltaro I have a question for you.

Some unittests are failing with assertion errors, such as `test/python/WMCore_t/WorkQueue_t/WorkQueue_t.py:WorkQueueTest.testProcessing`. We get something like

```plaintext
Traceback (most recent call last):
  File "/home/dmwm/wmcore_unittest/WMCore/test/python/WMCore_t/WorkQueue_t/WorkQueue_t.py", line 502, in testProcessing
    self.assertEqual(processedBlocks, 17)
AssertionError: 16 != 17
```

The problem seems to be that when from here

https://github.com/dmwm/WMCore/blob/ea0cb37f1a10453ebdaad1116bc4dc7627b2c6ff/src/python/WMCore/WorkQueue/WorkQueueBackend.py#L351

we call `CMSCouch.py:Database.loadList()`, its return value here

https://github.com/dmwm/WMCore/blob/ea0cb37f1a10453ebdaad1116bc4dc7627b2c6ff/src/python/WMCore/Database/CMSCouch.py#L586

contains different data every time. This function makes a call to `http://localhost:6994/workqueue_t/_design/WorkQueue/_list/workRestrictions/availableByPriority`.

This unstable behavior happens in py3 only, in py2 it always seems to return the same data. 

I have to questions:
- I am kind of lost here, could you point me to who is serving `http://localhost:6994/workqueue_t/_design/WorkQueue/_list/workRestrictions/availableByPriority` ?
- is this even a problem or we should mark these as unstable unittests for the time being, and get back to this at a later stage?

--- a/src/python/WMCore/Database/CMSCouch.py
+++ b/src/python/WMCore/Database/CMSCouch.py
@@ -70,7 +70,7 @@ class Document(dict):
         # https://issues.apache.org/jira/browse/COUCHDB-1141
         deletedDict = {'_id': self['_id'], '_rev': self['_rev'], '_deleted': True}
         self.update(deletedDict)
-        for key in self.keys():
+        for key in list(self.keys()):
             if key not in deletedDict:
                 del self[key]
 

--- a/src/python/WMCore/MicroService/MSTransferor/Workflow.py
+++ b/src/python/WMCore/MicroService/MSTransferor/Workflow.py
@@ -357,7 +357,7 @@ class Workflow(object):
             return [thisChunk], [thisChunkSize]
 
         # create a descendant list of blocks according to their sizes
-        sortedPrimary = sorted(viewitems(self.getPrimaryBlocks()), key=operator.itemgetter(1), reverse=True)
+        sortedPrimary = sorted(viewitems(self.getPrimaryBlocks()), key=lambda item: item[1]['blockSize'], reverse=True)
         if len(sortedPrimary) < numChunks:
             msg = "There are less blocks than chunks to create. "
             msg += "Reducing numChunks from %d to %d" % (numChunks, len(sortedPrimary))

--- a/test/etc/UnstableTests.txt
+++ b/test/etc/UnstableTests.txt
@@ -31,4 +31,6 @@ WMCore_t.WMBS_t.JobSplitting_t.EventBased_t.EventBasedTest:test100EventMultipleS
 WMCore_t.WMBS_t.JobSplitting_t.FileBased_t.FileBasedTest:testSiteWhiteBlacklist
 WMCore_t.WMBS_t.JobSplitting_t.RunBased_t.EventBasedTest:testMultipleRunsCombine
 WMCore_t.WMBS_t.JobSplitting_t.FileBased_t.FileBasedTest:testLocationSplit
+WMCore_t.WMSpec_t.StdSpecs_t.Express_t.ExpressTest:testFilesets
 WMCore_t.WMSpec_t.StdSpecs_t.ReReco_t.ReRecoTest:testFilesets 
+WMCore_t.WorkQueue_t.Policy_t.Start_t.Block_t.BlockTestCase:testPileupData

